OfBorg [says](https://logs.nix.ci/?attempt_id=f46f3118-1670-49b7-902c-87a831a486d0&key=nixos%2Fnixpkgs.129669) that the first python3-minimal step still doesn't build:
```
clang -L/nix/store/k69r3dlxjcihncggi4yxr2ljcj1cyf1j-zlib-1.2.11/lib -L/nix/store/szg19c03gyy5fjyclhdcwyi6dc9x9gys-bzip2-1.0.6.0.2/lib -L/nix/store/nl6s07d0nlc52as2aq2vzkjk7z9m4d2z-expat-2.4.1/lib -L/nix/store/b1irkxbsflwj44mb3h79ysbfhzp969zs-xz-5.2.5/lib -L/nix/store/i5x98h09cl8ckklf3brylwri6l43czlx-libffi-3.3/lib -L/nix/store/k69r3dlxjcihncggi4yxr2ljcj1cyf1j-zlib-1.2.11/lib -L/nix/store/szg19c03gyy5fjyclhdcwyi6dc9x9gys-bzip2-1.0.6.0.2/lib -L/nix/store/nl6s07d0nlc52as2aq2vzkjk7z9m4d2z-expat-2.4.1/lib -L/nix/store/b1irkxbsflwj44mb3h79ysbfhzp969zs-xz-5.2.5/lib -L/nix/store/i5x98h09cl8ckklf3brylwri6l43czlx-libffi-3.3/lib   -framework CoreFoundation -o python.exe Programs/python.o -L. -lpython3.9 -ldl    -framework CoreFoundation
DYLD_LIBRARY_PATH=/private/tmp/nix-build-python3-minimal-3.9.5.drv-0/Python-3.9.5 ./python.exe -E -S -m sysconfig --generate-posix-vars ;\
if test $? -ne 0 ; then \
        echo "generate-posix-vars failed" ; \
        rm -f ./pybuilddir.txt ; \
        exit 1 ; \
fi
dyld: lazy symbol binding failed: Symbol not found: ___isOSVersionAtLeast
  Referenced from: /private/tmp/nix-build-python3-minimal-3.9.5.drv-0/Python-3.9.5/libpython3.9.dylib
  Expected in: flat namespace
dyld: Symbol not found: ___isOSVersionAtLeast
  Referenced from: /private/tmp/nix-build-python3-minimal-3.9.5.drv-0/Python-3.9.5/libpython3.9.dylib
  Expected in: flat namespace
/nix/store/6jysbyhc43sjvfiyh1bpvi1n3zbz212r-bootstrap-tools/bin/bash: line 5: 10134 Abort trap: 6           DYLD_LIBRARY_PATH=/private/tmp/nix-build-python3-minimal-3.9.5.drv-0/Python-3.9.5 ./python.exe -E -S -m sysconfig --generate-posix-vars
generate-posix-vars failed
make: *** [Makefile:619: pybuilddir.txt] Error 1
builder for '/nix/store/dbv6f2r28griyb7gmy18kmz9kg7lq9mh-python3-minimal-3.9.5.drv' failed with exit code 2
```

--- a/pkgs/development/interpreters/python/cpython/default.nix
+++ b/pkgs/development/interpreters/python/cpython/default.nix
@@ -40,7 +40,8 @@
 , static ? stdenv.hostPlatform.isStatic
 , enableOptimizations ? false
 # enableNoSemanticInterposition is a subset of the enableOptimizations flag that doesn't harm reproducibility.
-, enableNoSemanticInterposition ? true
+# clang starts supporting `-fno-sematic-interposition` with version 10
+, enableNoSemanticInterposition ? (!stdenv.cc.isClang || (stdenv.cc.isClang && lib.versionAtLeast stdenv.cc.version "10"))
 # enableLTO is a subset of the enableOptimizations flag that doesn't harm reproducibility.
 # enabling LTO on 32bit arch causes downstream packages to fail when linking
 , enableLTO ? stdenv.is64bit

